### ISSUE TITLE
Not Work Ubuntu 16.04

### ISSUE THREAD

**Issue by hznovich** [fireice-uk/xmr-stak-amd#5]
Hi, not work in ubuntu 16.04 ((( Help please.
[2017-03-07 14:57:48] : Compiling code and initializing GPUs. This will take a while...
[2017-03-07 14:57:48] : Device 0 work size 8 / 256.
[2017-03-07 14:57:48] : Error CL_INVALID_BUFFER_SIZE when calling clCreateBuffer to create hash scratchpads buffer.

Ubuntu 16.04, AMD Radion HD 7850 2G RAM

startup script: 
```
#!/bin/bash
export GPU_FORCE_64BIT_PTR=1
export GPU_USE_SYNC_OBJECTS=1
export GPU_MAX_ALLOC_PERCENT=100
export GPU_SINGLE_ALLOC_PERCENT=100
export GPU_MAX_HEAP_SIZE=100
./xmr-stak-amd
```
config.txt:
```
/* 
 * Number of GPUs that you have in your system. Each GPU will get its own CPU thread.
 */
"gpu_thread_num" : 1,

/*
 * GPU configuration. You should play around with intensity and worksize as the fastest settings will vary.
 *      index    - GPU index number usually starts from 0
 *  intensity    - Number of parallel GPU threads (nothing to do with CPU threads)
 *   worksize    - Number of local GPU threads (nothing to do with CPU threads)
 * affine_to_cpu - This will affine the thread to a CPU. This can make a GPU miner play along nicer with a CPU miner.
 */
"gpu_threads_conf" : [ 
	{ "index" : 0, "intensity" : 1000, "worksize" : 8, "affine_to_cpu" : false },
],

/*
 * Platform index. This will be 0 unless you have different OpenCL platform - eg. AMD and Intel.
 */
"platform_index" : 0,
```

**Comment by fireice-uk** [comment]
Can you post the output of clinfo?

**Comment by hznovich** [comment]
Thanks. Works. What does this option mean?

**Comment by fireice-uk** [comment]
Amount of parallel hashes. Each CN hash takes up 2MB.

**Comment by gsedej** [comment]
May I ask what opencl do you use for HD 7850 on 16.04. Clover? Does it work? (I have one old r9 270 at home)

**Comment by hznovich** [comment]
Yes it works. A little later I will write.

**Comment by gsedej** [comment]
I figured it out.
The amdgpu-pro 17.10 (and 16.40) DOES work on R9 270 on ubuntu 16.04.2 (kernel 4.4). (regardless of no official support)

The installation is as simple as downloading driver (`amdgpu-pro-17.10-429170.tar.xz`) unpacking and running the `amdgpu-pro-install`.

The xmr-stak-amd works when you put the `export GPU...` lines
My settings are:
```
"gpu_threads_conf" : [ 
{ "index" : 0, "intensity" : 750, "worksize" : 10, "affine_to_cpu" : false },
...
"pool_address" : "xmr.crypto-pool.fr:3333",
],
```
gives me result
```
HASHRATE REPORT
| ID |   10s |   60s |   15m |
|  0 | 333.5 | 333.6 | 333.5 |
```
btw, i am totaly new to mining. I did create monero wallet and made monerod running. I have my "address", how to mine? which pool to use? Do i need to "register"?

**Comment by futurepr0n** [comment]
@gsedej what do you mean, export GPU - where do you put that? Im having the same issue. But i dont understand fro, your comment. Its a bit vague. Every entry in the config.txt needs to be proceeded by export???

**Comment by gsedej** [comment]
the "export lines" are environment variables and are put in the terminal. For linux you put "export VARIABLE_NAME=value" for windows I think it's "xset VARIABLE_NAME=value", but not sure.

The variable are in the original comment under "startup script:"

I don't think you can put this in config.txt. You would need to make system script (bash for linux, .bat for windows iirc) if you don't want to write down this all the time.

**Comment by futurepr0n** [comment]
Managed to apply this ahead of your response - apologies, but thank you for reaching out again! It was from an ethereum mining try - that I read that windows requires those exports but that linux did not. In any event, once I created a mine.sh script that used the exports from the original comment and then it runs ./xmr-stak-amd ../config.txt -- everything is golden. Thank you so much! Happy to contribute a lil computing power to the project.
